### ISSUE TITLE
Feature request: scan from stored image

### ISSUE THREAD

**Issue by porst17** [Fr4gorSoftware/SecScanQR#188]
Thank you for creating SecScanQR. It's a great app I use often.

Sometimes I need to scan QR codes that are part of photos I took earlier or screenshots from other apps. I am still having another proprietary QR scanning app on my phone just to do that.

The UI in that app is as follows, which I think would already be good enough:
- Click on "Scan image"
- Select a photo from the gallery
- Resize an on-screen rectangle until a code is detected (also very useful if the image/photo contains more than one code)
- Press "OK"

An extension to that UI would be to do some trapezoid correction instead of just selecting a rectangle. This would help with photos where the QR code is not properly aligned with the image plane.

I'd really appreciate if you would consider adding this useful feature to SecScanQR.

**Comment by michaelprager** [comment]
This feature already exists. It's a bit hidden because you have to share the image with the app, there is no dedicated file selection dialog.

In case anyone is interested: I've played around with adding a simple file selection dialog on a test branch here: https://github.com/michaelprager/SecScanQR/commit/2f40674eee710305fffd31d3d49fa046e8404397
It's not perfect though. For one, it adds READ_EXTERNAL_STORAGE because that is needed if a non stock file explorer app is installed on the system. Also the scan file button is only available when you press "scan" and then go back.

**Comment by Fr4gorSoftware** [comment]
@michaelprager is right but I will take a look at your version as a base to implement this feature in the UI of the app

**Comment by porst17** [comment]
Thanks for pointing me to the sharing functionality. I didn't notice that feature before. This already helps, but has its limitations. I tested it a bit and the results are mixed. I'll report here, but maybe this should be moved to a separate issue (specific to the "Share" functionality):
- QR codes generated by the app itself are properly recognized and many I found on the web.
- QR codes that are only mildly processed are often not recognized. I got one that was send via email as a JPEG and the slight blurring and reduction in quality seemed to break the recognition when "sharing" the image with the app. Opening the email on my computer and scanning it from the screen via SecScanQR worked immediately. I could reproduce this problem using a QR code generated by the app, opening it in an image editor, resizing it (such that the image is very slightly blurred) and saving it again. 
- I was not able to recognize QR codes that are part of photos are screen shots this way.
- I don't really understand under which circumstances I can share images to the app for scanning. SecScanQr is available for sharing via the photo gallery, but not e.g. when I share images from the web browser (Firefox, in my case). When I try to "Share image" from the web browser, SecScanQR is listed as "Barcode Generator" and "QR-Code generator", but it's not available for scanning.

So, unfortunately, the way it's currently implemented isn't of much use due to the many cases where the recognition doesn't work.

**Comment by UsrPwndUX** [comment]
It doesn't seem to work with sharing an image of a QR code to the app. I have some success with some barcodes sharing to app. It would be great to have the ability to click scan in the app then select the image you want to scan.

**Comment by michaelprager** [comment]
> It doesn't seem to work with sharing an image of a QR code to the app. I have some success with some barcodes sharing to app.

Yes you are right, sharing only worked for barcodes but not QR codes. This has been fixed already in the code, see #184. This should work in the next release.

**Comment by Kache** [comment]
Just found this issue looking for the feature, myself. Perhaps an update to the request to:

Make this feature more discoverable by adding instructions in the app?

There's a decent amount of whitespace on the landing view for even just raw text.

Incremental improvements can follow later, e.g. embed the "share" icon in instructions, visual walkthrough, etc.
